### ISSUE TITLE
Problem z generowanie klas na podstawie pliku swagger.yaml (sale/product-offers)

### ISSUE THREAD

**Issue by ryszardlet** [allegro/allegro-api#9407]
Wygenerowałem klasy potrzebne do komunikacji z API na podstawie wspomnianego pliku. Niestety jak się okazuje - po wygenerowaniu klas nie mam wszystkich opcji (parametrów). Chodzi oczywiście o https://developer.allegro.pl/documentation#operation/createProductOffers. Za to pojawiły się 2 klasy _SaleProductOfferRequestV1_ oraz _SaleProductOfferRequestV1AllOf_ . Jak można to obsłużyć?

**Comment by Lukasz-Zurek** [comment]
Weryfikuję ten temat z zespołem odpowiedzialnym za zasób. Proszę o cierpliwość.

**Comment by ryszardlet** [comment]
> Weryfikuję ten temat z zespołem odpowiedzialnym za zasób. Proszę o cierpliwość.

Czy może udało się Państwu coś ustalić?

**Comment by Lukasz-Zurek** [comment]
@ryszardlet Tak. Problemy z generowaniem klas pod openapi-generator w wersjach 6+ (ale i nie tylko) - były nam znane już wcześniej i tutaj niestety mogę jedynie potwierdzić, że _SaleProductOfferRequestV1AllOf_ nadal nie działa poprawnie. Problemy mogą występować również dla _SaleProductOfferRequestV1_. Nie wspieramy w pełni generatora i nie przewidujemy tutaj w najbliższym czasie optymalizacji w tym zakresie, ponieważ konieczne byłyby poważniejsze zmiany, wymagające czasu. Niemniej jednak Twoje zgłoszenie jest dla nas pewnym sygnałem do dalszych dyskusji. 

Jeżeli na naszym forum są inni użytkownicy, którzy mają podobne problemy, lub udaje im się z nimi radzić np. na starszych wersjach openapi-generator - poproszę o informację w tym wątku. Wszelkie uwagi mile widziane.

**Comment by ryszardlet** [comment]
Ok, rozumiem. Jakiego zatem generatora użyć, żeby nie mieć takich problemów?

**Comment by Lukasz-Zurek** [comment]
@ryszardlet Prawdopodobnie pod starszymi wersjami openapi-generator np. 5.4. i niżej - powinno to działać prawidłowo. Oficjalnie jednak nie wspieramy w pełni generatora, dlatego zaapelowałem tutaj do pozostałych użytkowników forum - być może znajdzie się ktoś, kto poradził sobie z tym problemem.
